blockservice/blockservice.go

```diff
@@ -199,12 +206,14 @@ func getBlock(ctx context.Context, c *cid.Cid, bs blockstore.Blockstore, f excha
 // the returned channel.
 // NB: No guarantees are made about order.
 func (s *blockService) GetBlocks(ctx context.Context, ks []*cid.Cid) <-chan blocks.Block {
+	ctx = log.EventBeginInContext(ctx, "BlockService.GetBlocks")
 	return getBlocks(ctx, ks, s.blockstore, s.exchange)
 }
 
 func getBlocks(ctx context.Context, ks []*cid.Cid, bs blockstore.Blockstore, f exchange.Fetcher) <-chan blocks.Block {
 	out := make(chan blocks.Block)
 	go func() {
+		defer logging.MaybeFinishEvent(ctx)
```

Comment:
[This PR](https://github.com/ipfs/go-log/pull/8) needs to go first for this functionality.